test/e2e/specs/editor/blocks/site-title.spec.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'Site Title block', () => {
+	let originalSiteTitle;
+
+	test.beforeAll( async ( { requestUtils } ) => {
+		originalSiteTitle = ( await requestUtils.getSiteSettings() ).title;
+	} );
+
+	test.afterAll( async ( { requestUtils } ) => {
+		await requestUtils.updateSiteSettings( { title: originalSiteTitle } );
+	} );
+
+	test( 'Can edit the site title as admin', async ( {
+		admin,
+		editor,
+		page,
+	} ) => {
+		await admin.createNewPost();
+		await editor.insertBlock( { name: 'core/site-title' } );
+
+		const siteTitleBlock = editor.canvas.getByRole( 'document', {
+			name: 'Block: Site Title',
+		} );
+
+		// Update the site title
+		await siteTitleBlock
+			.getByRole( 'textbox', {
+				name: 'Site title text',
+			} )
+			.fill( 'New Site Title' );
+
+		await editor.publishPost();
+		await page.reload();
+
+		await expect( siteTitleBlock ).toBeVisible();
+		await expect( siteTitleBlock ).toHaveText( 'New Site Title' );
+	} );
+
+	// Reason: Reason: The current e2e test setup doesn't provide an easy way to switch between user roles.
+	// eslint-disable-next-line playwright/expect-expect
+	test.fixme( 'Cannot edit the site title as editor', async () => {} );
```

Comment:
I tried code like below but it didn't work 😅 We can create a user, but it's difficult to switch to that user at the moment, right?

```javascript
test( 'Cannot edit the site title as editor', async ( {
	requestUtils,
	admin,
	page,
} ) => {
	await requestUtils.createUser( {
		username: 'testuser',
		email: 'testuser@example.com',
		firstName: 'Jane',
		lastName: 'Doe',
		password: 'secret',
	} );
	await page.reload();
	await admin.createNewPost();
	// ...
	await requestUtils.deleteAllUsers();
} );
```

Reply:
That’s the reason this test is unfinished. We need to update our utility methods to support multiple roles.